test/amqp-integration-test.py

```diff
@@ -36,7 +36,7 @@ def __init__(self, cmd, proc):
 def run(path):
     global processes
     binary = os.path.join(tempdir, os.path.basename(path))
-    cmd = 'go build -tags pkcs11 -o %s %s' % (binary, path)
+    cmd = 'GORACE="halt_on_error=1" go build -tags pkcs11 -race -o %s %s' % (binary, path)
```

Comment:
You make a good point! The current version looks a little messy but does have that advantage.